### PR TITLE
add faq documentation

### PR DESCRIPTION
**Pull request by mx2323**
documentation

### CODE REVIEW

File: Documentation/faq.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+In situations where a client connects to a minority, etcd
+favors by default availability over consistency. This means that even though
+data might be “out of date”, it is still better to return something versus
+nothing. 
+
+In order to confirm that a read is up to date with a majority of the cluster,
+the client can use the `quorum=true` parameter on reads of keys. This means
+that a majority of the cluster is checked on reads before returning the data,
+otherwise the read fails.
```
Comment (xiang90) [inline Documentation/faq.md]: the read will timeout and fail.

File: Documentation/faq.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## 2) With quorum=false, doesn’t this mean that if my client switched the member it was connected to, that it could experience a logical ordering where the cluster goes backwards in time? 
+
+Yes, but this could be handled at the etcd client implementation via
+remembering the last seen index. The “index” is the cluster's single
+irrevocable sequence of the entire  modification history. The client could
```
Comment (xiang90) [inline Documentation/faq.md]: there is an extra space between entire and modification history.

File: Documentation/faq.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+The watch will stay untriggered, even as modifications are occurring in the
+majority quorum. This is an open issue, and is being addressed in v3. There are
+multiple ways to work around the watch trigger not firing. 
+
+1) build a signaling mechanism independent of etcd. This could be as simple as
+a “pulse” to the client to reissue a GET with quorum=true for the most recent
+version of the data. 
+   
+2) poll on the `/health` endpoint, and verify that the local member’s `/health`
```
Comment (xiang90) [inline Documentation/faq.md]: you can get on `/v2/keys/` and check the `raft-index` is increasing every 0.5 second.

File: Documentation/faq.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+The design goal of etcd is that reconfiguration is simply an API, and health
+monitoring and addition/removal of members is up to the individual application
+and their integration with the reconfiguration API. 
+
+Thus, a member that is down, even infinitely, will never be automatically
+removed from the etcd cluster member list. 
+
+This makes sense because its usually an application level / administrative
+action to determine whether a reconfiguration should happen based on health. 
+
```
Comment (xiang90) [inline Documentation/faq.md]: can we add a link to https://github.com/coreos/etcd/blob/master/Documentation/runtime-reconf-design.md?

**Comment by xiang90** [comment]
Overall, this looks good to me. We probably need to make its format matches other doc (for example: https://github.com/coreos/etcd/blob/master/Documentation/implementation-faq.md). Actually, we should also move this to `implementation-faq.md`? 

/cc @yichengq @robszumski Can you please also take a look?

File: Documentation/faq.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+proxy specify both the `--proxy` flag and the `--initial-cluster` flag. 
+
+From there, any etcdctl client that starts up automatically speaks to the local
+proxy and the proxy redirects operations to the current configuration of the
+cluster it was originally paired with. 
+
+In the v2 spec of etcd, proxies cannot be promoted to members of the cluster.
+They also cannot be promoted to followers or at any point become part of the
+replication of the etcd cluster itself. 
+
```
Comment (robszumski) [inline Documentation/faq.md]: Might be good to mention that you can add/remove members, but not from a proxy:

"etcd supports reconfiguration of the cluster (without downtime) to add or remove members of the cluster. You can follow the [runtime reconfiguration docs]() for the available options."

File: Documentation/faq.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+## 5) How is cluster membership and health handled in etcd v2? 
+
+The design goal of etcd is that reconfiguration is simply an API, and health
+monitoring and addition/removal of members is up to the individual application
+and their integration with the reconfiguration API.
```
Comment (robszumski) [inline Documentation/faq.md]: Link to the reconfiguration docs

File: Documentation/faq.md (hunk 5 above)
Comment (robszumski) [inline Documentation/faq.md]: Whoops, you mentioned this right below. Still might make sense to mention here too.

**Comment by robszumski** [comment]
Thanks for contributing, these are great questions!

File: Documentation/faq.md (hunk 5 above)
Comment (mx2323) [inline Documentation/faq.md]: Might be good to mention that you can add/remove members, but not from a proxy: <-- is this true? 

i believe member add/remove can be done from a proxy, just that there is no concept of adding and removing proxies from a cluster.

**Comment by xiang90** [comment]
@mx2323 Can you please squash all the commits into one? I will go ahead and merge this. I will do further cleanup to make this match the format style of other docs.

**Comment by mx2323** [comment]
made all changes, except didn't merge implementation-faq.md with this faq. this faq is intended to be short, and go over consistency behaviors. implementation-faq.md looks like its answering a single question, and is quite long.

**Comment by xiang90** [comment]
@mx2323 Sorry. I should mention before. The commit message should be in the format of "subsystem: message". In this case, it should be `doc: add faq documentation`.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**